packages/bruno-electron/src/ipc/network/index.js

```diff
@@ -232,6 +233,11 @@ const configureRequest = async (
         request.data = passwordData;
         request.url = passwordAccessTokenUrl;
         break;
+      case 'implicit':
+        interpolateVars(requestCopy, envVars, collectionVariables, processEnvVars);
+        const { accessToken } = await getOAuth2ImplicitToken(requestCopy, collectionUid);
+        request.headers['Authorization'] = `Bearer ${accessToken}`;
+        break;
```

Comment:
@lohxt1 , please help me understand this part - the `request` here is what user configured in the client, and we only want to add the authorization to it. Why do all other grand types modify the request's method, or data, and not simply add ```request.headers['Authorization'] = `Bearer ${accessToken}`;```?

Am I completly missing the point of what's going on, or is this a bug in the implementation of all other grant types?

Reply:
@lohxt1 
I would expect that requests to auth url and token url are separate from main collection request. Current behaviour is to overwrite information about original user's request and leads to bugs such as this: https://github.com/usebruno/bruno/issues/2060 .

I thought maybe somewhere in this file the request is copied, repurposed as authorization request,  executed, logged in Timeline tab and used to extract access_token which is then added to original request exactly as specified by the user, but it is not happening - the code here leaves the request modified and original one is lost. 

Now, it cannot be the desired outcome, can it?

Reply:
It would make sense to create a predefined and documented variable for token data and allow to reference it from pre/post/collection level scripts. It would for example store:
`access_token`, `refresh_token`, `token_type`, `expires_in` and `scope` and live in collection level. It would offer the same scripting flexibility, but also allow for more intuitive UI (resolving referenced bugs) and serve as a starting point for implementing any logic for automatic token management (e.g. automaic attempts to use refresh token when access_token is expired or skip token request if one is already stored). 
With this approach Implicit Flow would not be the odd one out because it cannot even return access_token in a response body ;). Not sure however how to best make it work.